### ISSUE TITLE
Controlled TextField with required enabled initially renders invalid

### ISSUE THREAD

**Issue by robcaldecottvelo** [jamesmfriedman/rmwc#280]
I am using RMWC 1.8.3

If you have a controlled TextField with **required** set it will initially render as invalid due to the initial syncing of the React component props to the MDC component internal state. However, required fields should not render as invalid until they lose focus or the form is submitted.

For example, the following code will incorrectly render the second text field as invalid because the initial value is en empty string:

```js
import React from "react";
import ReactDOM from "react-dom";
import "material-components-web/dist/material-components-web.min.css";
import "typeface-roboto";
import { TextField } from "rmwc/TextField";
import { Button } from "rmwc/Button";

class App extends React.Component {
  state = { name: "", email: "" };

  render() {
    return (
      <form
        onSubmit={e => {
          e.preventDefault();
        }}
      >
        <TextField
          label="Name"
          value={this.state.name}
          onChange={e => {
            this.setState({ name: e.target.value });
          }}
        />

        <TextField
          required
          label="Email"
          value={this.state.email}
          onChange={e => {
            this.setState({ email: e.target.value });
          }}
        />

        <Button type="submit">Submit</Button>
      </form>
    );
  }
}

ReactDOM.render(<App />, document.getElementById("root"));
```

Expected behaviour would be for the email TextField to render as valid until the user tabs in/out of it or the submit button is clicked.

To work around this I am using a combination of the TextField `invalid` prop and `onBlur` and handling this myself but if there are other suggestions they would be welcome.

**Comment by jdeanwaite** [comment]
Seeing this as well.

**Comment by jamesmfriedman** [comment]
This was fixed with the release of 1.9 since MDC overhauled their internal validation checks. Please reopen if you are still seeing issues.
